src/dict/dsl_details.cc

```diff
@@ -832,7 +832,34 @@ DslScanner::DslScanner( string const & fileName ):
   bool needExactEncoding = false;
 
   QByteArray ba = QByteArray::fromRawData( (const char *)firstBytes, 50 );
-  codec         = QTextCodec::codecForUtfText( ba, QTextCodec::codecForName( "UTF-8" ) );
+  codec         = QTextCodec::codecForUtfText( ba, nullptr );
+  if ( !codec ) {
+    // the encoding has no bom.
+    // check the first char # (0x23).
+    auto hashTag      = 0x0023;
+    const uchar * buf = reinterpret_cast< const uchar * >( ba.constData() );
+
+    uint uci = qFromUnaligned< uint >( buf );
```

Comment:
`uint` and `ushort` are not in either C or C++'s standards. Maybe replace them with`uint64_t`,`uint32_t` from `<cstdint>`?